File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *a, ListNode *b) {
        ListNode * p1 = a;
        ListNode * p2 = b;
        while(p1 and p2){
            ListNode * p2 = b;
            while(p2){
                if(p1 == p2) return p1;
                p2 = p2->next;
            }
            p1 = p1->next;
        }
        
        return NULL;
    }
};

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, a: ListNode, b: ListNode) -> Optional[ListNode]:
        cntA, cntB = 0, 0
        tmp = a
        while tmp:
            cntA = cntA+1
            tmp = tmp.next
        tmp = b
        while tmp:
            cntB = cntB+1
            tmp = tmp.next
        ptrA = a
        ptrB = b
        if cntA > cntB:
            tmp = a
            while cntA-cntB:
                tmp = tmp.next
                cntA-=1
            ptrA = tmp
        else: 
            tmp = b
            while cntB-cntA:
                tmp = tmp.next
                cntB-=1
            ptrB = tmp
        
        while ptrA and ptrB and ptrA != ptrB:
            ptrA = ptrA.next
            ptrB = ptrB.next
            
        return ptrA